git-cz.js throwing error. Commit still goes through

Getting the following error after running git cz. version commitizen@2.8.4

```
/node_modules/commitizen/dist/cli/strategies/git-cz.js:102
        throw error;
```

Commit seems to go through, however error doesn't provide much more details.

@marionebl I am seeing the same thing. My info is below. It doesn't look too helpful though. I have to call it a night, but if I have time later, I'll make the change locally to include `stderr` and look for more info.

* OS: OS X El Capitan (10.11.6)
* node: v6.3.1
* npm: 3.10.3
* adapter: cz-conventional-changelog
* command: simply `git-cz`

Error output:
```
/Users/brandon/src/most-proxy/node_modules/commitizen/dist/cli/strategies/git-cz.js:102
        throw error;
        ^
fix(dependencies): Add most.js to devDependencies

most.js is a development dependency of most-proxy and is now automatically installed with the rest
of the development dependencies.



npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/6.3.1/bin/node" "/usr/local/bin/npm" "run" "commit"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! most-proxy@3.0.0 commit: `git-cz`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the most-proxy@3.0.0 commit script 'git-cz'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the most-proxy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     git-cz
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs most-proxy
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls most-proxy
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/brandon/src/most-proxy/npm-debug.log
```

--- a/src/cli/strategies/git-cz.js
+++ b/src/cli/strategies/git-cz.js
@@ -62,15 +62,14 @@ function gitCz(rawGitArgs, environment, adapterConfig) {
     console.log(`cz-cli@${cliPackageJson.version}, ${adapterPackageJson.name}@${adapterPackageJson.version}\n`);
     commit(sh, inquirer, process.cwd(), prompter, {
       args: parsedGitCzArgs,
-      disableAppendPaths:true,
-      emitData:true,
-      quiet:false,
+      disableAppendPaths: true,
+      emitData: true,
+      quiet: false,
       retryLastCommit
     }, function(error) {
       if (error) {
         throw error;
       }
-      // console.log('commit happened');
     });
   });
 

--- a/src/commitizen/commit.js
+++ b/src/commitizen/commit.js
@@ -1,4 +1,5 @@
 import path from 'path';
+
 import homeOrTmp from 'home-or-tmp';
 import dedent from 'dedent';
 import {commit as gitCommit, log} from '../git';
@@ -10,12 +11,10 @@ export default commit;
  * Takes all of the final inputs needed in order to make dispatch a git commit
  */
 function dispatchGitCommit(sh, repoPath, template, options, overrideOptions, done) {
-  
     // Commit the user input -- side effect that we'll test
-    gitCommit(sh, repoPath, template, { ...options, ...overrideOptions }, function() {
-      done(template);
+    gitCommit(sh, repoPath, template, { ...options, ...overrideOptions }, function(error) {
+      done(error, template);
     });
-    
 }
 
  /**
@@ -31,7 +30,7 @@ function commit(sh, inquirer, repoPath, prompter, options, done) {
     
     // We want to use the last commit instead of the current commit,
     // so lets override some options using the values from cache.
-    let { 
+    let {
       options: retryOptions,
       overrideOptions: retryOverrideOptions,
       template: retryTemplate
@@ -40,12 +39,23 @@ function commit(sh, inquirer, repoPath, prompter, options, done) {
     
   } else {
     // Get user input -- side effect that is hard to test
-    prompter(inquirer, function(template, overrideOptions) {
-    
+    prompter(inquirer, function(error, template, overrideOptions) {
+      // Allow adapters to error out
+      // (error: Error?, template: String, overrideOptions: Object)
+      if (!(error instanceof Error)) {
+        overrideOptions = template;
+        template = error;
+        error = null;
+      }
+
+      if (error) {
+        return done(error);
+      }
+
       // We don't want to add retries to the cache, only actual commands
       cache.setCacheValueSync(cachePath, repoPath, { template, options, overrideOptions });
       dispatchGitCommit(sh, repoPath, template, options, overrideOptions, done);
-    }); 
+    });
   }
    
 }

--- a/src/git/commit.js
+++ b/src/git/commit.js
@@ -31,6 +31,7 @@ function commit(sh, repoPath, message, options, done) {
     stdio: options.quiet ? 'ignore' : 'inherit'
   }, function(error, stdout, stderror) {
     if (error) {
+      error.message = [error.message, stderror].filter(Boolean).join('\n');
       return done(error);
     }
     done();

